src/pybel/manager/cache_manager.py

```diff
@@ -672,8 +685,7 @@ def drop_network_by_id(self, network_id):
     def drop_networks(self):
         """Drops all networks"""
         for network in self.session.query(Network).all():
-            self.session.delete(network)
-            self.session.commit()
+            self.drop_network(network)
```

Comment:
There might be a more direct way to do this since it just involves dropping all edges, then all networks (no worry about which is associated with which)

Reply:
As I think you found out, I switched to this because of problems with database integrity violations. Make it run, make it correct, then make it fast. I've only made it to step 2. :)

Reply:
The optimization for this would be having drop_network use a list of network id's and then instead of all the searches using `network_id == network.id` you'd do `network_id.in_(network_ids)`.

The next optimization is `TRUNCATE`.

Either way I think deleting everything is infrequent enough that this probably doesn't matter.